src/obstacle_detection_node.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python
+
+# ==============================================================================
+# === Imports =================================================================
+# ==============================================================================
+import copy
+import math
+from threading import Lock
+
+import cv2
+import numpy as np
+import rospy
+import tf
+import tf2_ros
+from cv_bridge import CvBridge
+from geometry_msgs.msg import Point, Pose, PoseArray, PoseStamped, Quaternion
+from sensor_msgs.msg import CameraInfo, Image
+
+from constants import *
+
+# ==============================================================================
+# === Global Variables =========================================================
+# ==============================================================================
+Q_MATRIX = None
+DISPARITY_MAP = None
+OBSTACLE_POINTS = None
+DRONE_POSE = None
+
+CAM_INFO1 = None
+CAM_INFO2 = None
+
+MAPX1 = None
+MAPY1 = None
+MAPX2 = None
+MAPY2 = None
+
+OBSTACLE_RADIUS = 0.1524 # meters
+
+BRIDGE = CvBridge()
+
+# ==============================================================================
+# === Helper Functions =========================================================
+# ==============================================================================
+def update_disparity_map(msg):
+    global DISPARITY_MAP
+    DISPARITY_MAP = msg
+
+def update_drone_pose(msg):
+    global DRONE_POSE
+    DRONE_POSE = msg
+
+def camera_info_cb1(msg):
+    global CAM_INFO1
+    CAM_INFO1 = copy.deepcopy(msg)
+
```

Comment:
we should add a check to see if cam_info is none. If it is not, we don't need to run this deep copy again since it should be constant